### PR TITLE
Create pull_request_template

### PR DESCRIPTION
**Pull request by askdba**
This is the first draft version of the pull request template.

### CODE REVIEW

File: pull_request_template [hunk 1]
```diff
@@ -0,0 +1,42 @@
+## Status
+**READY/IN DEVELOPMENT/HOLD**
```
Comment (shlomi-noach) [inline pull_request_template:2]: I'd like to suggest we use the formal "draft pull request" to indicate that a pull request is not ready; and other wise the popular convention of "WIP" in the title.

File: pull_request_template [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Backport
+YES | NO
+
+## Description
```
Comment (shlomi-noach) [inline pull_request_template:7]: Let's make this the very first section.

File: pull_request_template [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+A few sentences describing the overall goals of the pull request's commits.
+
+## Related PRs
```
Comment (shlomi-noach) [inline pull_request_template:10]: Let's change that to "Related Issue(s)" - we've already decided during our 1st retrospective that each PR must be associated with an issue.

File: pull_request_template [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+List related PRs against other branches:
+
+branch | PR
+------ | ------
+planetscale/vitess | [link]()
+vitessio/vitess [link]()
```
Comment (shlomi-noach) [inline pull_request_template:16]: IMO there is no need for this. You just put the link to an issue or PR and GitHub markdown automatically turns it into a link, with hover hints.

File: pull_request_template [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+
+## Todos
+- [ ] Tests
+- [ ] Documentation
```
Comment (shlomi-noach) [inline pull_request_template:21]: :+1:

File: pull_request_template (hunk 4 above)
Comment (shlomi-noach) [reply pull_request_template:16]: So, to clarify, the user will just put some links. Or even just `#34` -- GitHub will turn that into the correct link to issue 34.

File: pull_request_template [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+
+## Deploy Notes
+Notes regarding deployment the contained body of work.  These should note any
+db migrations, etc.
+
+## Steps to Test or Reproduce
+Outline the steps to test or reproduce the PR here.
```
Comment (shlomi-noach) [inline pull_request_template:29]: reproduction is for bugs, not for PRs, I think we should remove this section.

**Review by shlomi-noach — CHANGES_REQUESTED** [review]
Thanks for working on this @askdba ! I've listed some thoughts and changes.


let's move the file to `.github/PULL_REQUEST_TEMPLATE`

File: .github/PULL_REQUEST_TEMPLATE [hunk 7]
```diff
@@ -1,42 +1,27 @@
...
-1. 
-
 ## Impacted Areas in Vitess
 List general components of the application that this PR will affect:
 
-*
+* Query Serving
+* VReplication
+* Cluster Management
+* Build
```
Comment (shlomi-noach) [inline .github/PULL_REQUEST_TEMPLATE:27]:
```suggestion
- [ ] Query Serving
- [ ] VReplication
- [ ] Cluster Management
- [ ] Build 
```

**Review by shlomi-noach — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+## Related Issue(s)
+List related PRs against other branches:
+
+## Todos
+- [ ] Tests
+- [ ] Documentation
+
+## Deploy Notes
+Notes regarding deployment the contained body of work.  These should note any
```
Comment (deepthi) [inline .github/PULL_REQUEST_TEMPLATE:18]:
```suggestion
Notes regarding deployment of the contained body of work.  These should note any
```

**Review by deepthi — CHANGES_REQUESTED** [review]

**Review by deepthi — APPROVED** [review]

**Pull request merged**